README.md

```diff
@@ -179,142 +73,100 @@ impl MyContract<'_> {
         }
     }
 
-    #[sv::msg(instantiate)]
-    pub fn instantiate(&self, ctx: InstantiateCtx) -> StdResult<Response> {
-        self.counter.save(ctx.deps.storage, &0)?;
-
-        Ok(Response::new())
-    }
-}
-```
-
-We need to add this generic lifetime because of an optimization in storage plus -
-it doesn't want to take an owned string, as we often pass there a static string,
-but it also doesn't want to fix the `'static` ownership. 99% of the time, you can
-get away with passing just `'static` as the first `Item` generic argument, but I
-find it more convenient to introduce this "proxy" lifetime passing everywhere. I
-eliminate it in the `new` constructor, where I create the storage-plus accessors
-giving them proper keys.
-
-Now let's pass the initial counter state as a function argument:
-
-```rust
-#[contract]
-impl MyContract<'_> {
     #[sv::msg(instantiate)]
     pub fn instantiate(&self, ctx: InstantiateCtx, counter: u64) -> StdResult<Response> {
         self.counter.save(ctx.deps.storage, &counter)?;
-
         Ok(Response::new())
     }
-}
-```
-
-Sylvia would add the field into the instantiation message, which now becomes this:
-
-```rust
-struct InstantiateMsg {
-    counter: u64,
-}
-```
 
-What is essential - the field in the `InstantiateMsg` gets the same name as the
-function argument.
-
-Now let's add an execution message to the contract:
-
-```rust
-#[contract]
-impl MyContract<'_> {
     #[sv::msg(exec)]
-    pub fn increment(&self, ctx: ExecCtx) -> StdResult<Response> {
+    pub fn increment(&self, ctx: ExecCtx) -> Result<Response, ContractError> {
```

Comment:
Make it proper sylvia contract. If you switch to different error type add `sv::error` attribute to macro call